Optimize bulk actions endpoint & update gaps

denar50 opened the pull request:
## Summary
These optimizations aim to improve the performance of the bulk actions endpoint. 

The first optimization has to do with how we resolve rules before executing the bulk actions. Before this PR we were doing this sequentially, but a `bulkGet` method has been added to the rules action client.

The second optimization greatly improves the update gaps routine that occurs after a backfill is scheduled (see the before and after screenshots below).

The following screenshots were taken under these conditions:
I triggered a manual run bulk action on 1 rule (5m) with 1000 gaps over a period of 90 days.

Before:
![image](https://github.com/user-attachments/assets/44afc653-690c-4b04-b333-7b84faa19c25)

After:
![image](https://github.com/user-attachments/assets/31cec3f3-dd21-4e1e-a5e3-a957bbcbba03)


## How to test?
Use [this tool](https://github.com/elastic/security-documents-generator) to create 1 rule (5m) with 1000 gaps.
```
yarn start rules --rules 1 -g 1000 -c -i"5m"
```

Then do a manual run on it for a period of 90 days.

<!--ONMERGE {"backportTargets":["8.19"]} ONMERGE-->

denar50 committed:
Add bulk get rules to rules client
Use it in the bulk actions endpoint

denar50 committed:
Optimize the update gaps routine
- Pre process the scheduled items before going through the for each gap
- Only process scheduled items that overlap with the time range of the gap

elasticmachine commented:
Pinging @elastic/security-detection-engine (Team:Detection Engine)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/304000)
* Commit: b1cd382b8f16cf0646f4e74be09cdbb709cd75dd

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/304000#01972fe9-9fd4-4098-a34b-595a0b9e2111)

cc @denar50

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304000","number":304000,"commit":"b1cd382b8f16cf0646f4e74be09cdbb709cd75dd"}],"number":304000}
buildkite-pr-comment-->

ymao1 commented on x-pack/platform/plugins/shared/alerting/server/application/rule/methods/get/get_rules.ts:
```diff
@@ -0,0 +1,117 @@
+
+  const savedObjects: Awaited<ReturnType<typeof bulkGetRulesSo>>['saved_objects'] = [];
+
+  await pMap(
```
our other bulk functions use this function `checkAuthorizationAndGetTotal` to get the rules and perform the auth check, i wonder if we could reuse this function to perform all the auth checks and then do the bulkGetSo call afterward. i like having that logic in a centralized place used by all the bulk functions so if something about the auth check changes, we only have to update one spot.

the only difference being if any rule type is not authorized, the entire function would throw, vs this implementation where it would return partial results. however since all the detection rule types are covered under the same feature privilege, I imagine this is not a big concern for you all?

auth check function: https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/alerting/server/rules_client/lib/check_authorization_and_get_total.ts

example usage in bulk enable: https://github.com/elastic/kibana/blob/4bc129e17e483fda87f4e4ef0e466c91ba2389da/x-pack/platform/plugins/shared/alerting/server/application/rule/methods/bulk_enable/bulk_enable_rules.ts#L100

denar50 replied:
Done. I pushed a commit where I use the function that you suggested instead.

ymao1 commented on x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/update/utils.ts:
```diff
@@ -0,0 +1,116 @@
+
+export const toScheduledItem = (backfillSchedule: BackfillSchedule): ScheduledItem => {
+  const runAt = new Date(backfillSchedule.runAt).getTime();
+  const intervalDuration = parseDuration(backfillSchedule.interval);
```
`parseDuration` can throw an error, should there be an attempt to catch and handle?

ymao1 commented on x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/gap/interval_utils.ts:
```diff
@@ -209,17 +209,27 @@ export const denormalizeInterval = (interval: Interval): StringInterval => {
+  return { gte: clipped.start, lte: clipped.end };
+};
+
+export const clipDateIntervals = (
```
do we need any unit tests for this?

denar50 replied:
Unit test added 🙏

ymao1 reviewed (commented):
Code review only primarily focused on the rules client bulk get function.

denar50 commented on x-pack/platform/plugins/shared/alerting/server/application/rule/methods/get/schemas/get_rules_params_schema.ts:
```diff
@@ -0,0 +1,15 @@
+import { schema } from '@kbn/config-schema';
+
+export const getRulesParamsSchema = schema.object({
+  ids: schema.arrayOf(schema.string()),
```
Having `minLength` of 1 and a default value of an empty array seems a bit conflicting to me. How about we enforce `minLength` of 1 and throw an error if that is not the case as you suggested in the comment below?

ymao1 replied:
the `minLength` applies to the `schema.string()` while the `defaultValue: []` applies to the `schema.arrayOf`

denar50 committed:
Code review improvements
- move get rules to its own method folder in the rules client
- rename bulkGet to bulkGetRules
- move the transformRuleSoToSanitizedRule function to the transforms folder

denar50 commented on x-pack/platform/plugins/shared/alerting/server/rules_client/rules_client.ts:
```diff
@@ -170,6 +170,8 @@ export class RulesClient {
     getActionErrorLogWithAuth(this.context, params);
 
+  public bulkGet = <Params extends RuleTypeParams = never>(params: GetRulesParams) =>
+    getRules<Params>(this.context, params);
```
Done

denar50 commented on x-pack/platform/plugins/shared/alerting/server/application/rule/methods/get/get_rules.ts:
```diff
@@ -0,0 +1,117 @@
+  } catch (error) {
+    throw Boom.badRequest(`Error validating get rules params - ${error.message}`);
+  }
+
```
Done

denar50 committed:
use the same authorization logic that other bulk actions use

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306568)
* Commit: 1cd081e45c559a46db941163be901a94d93053f4

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/306568#0197547a-c995-4893-b9df-3cb59a3a5768)

### History
* :broken_heart: [Build #306527](https://buildkite.com/elastic/kibana-pull-request/builds/306527) failed 37ae9b88b481ac69689b5024ccdeb45e191fb7b7
* :green_heart: [Build #305932](https://buildkite.com/elastic/kibana-pull-request/builds/305932) succeeded 5dd6de38d86fb038a72cab32b62a28f38e7ef4a1
* :yellow_heart: [Build #305217](https://buildkite.com/elastic/kibana-pull-request/builds/305217) was flaky a51e110b4513825ad9df257c4ab1b2dab2592647
* :green_heart: [Build #305158](https://buildkite.com/elastic/kibana-pull-request/builds/305158) succeeded 6d41b31a555a75523f362208dee136f3febfc550
* :broken_heart: [Build #304509](https://buildkite.com/elastic/kibana-pull-request/builds/304509) failed 194e4286ce9fc8eb5b82229c6b6e44c6986f8f67

cc @denar50

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306568","number":306568,"commit":"1cd081e45c559a46db941163be901a94d93053f4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306527","number":306527,"commit":"37ae9b88b481ac69689b5024ccdeb45e191fb7b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305932","number":305932,"commit":"5dd6de38d86fb038a72cab32b62a28f38e7ef4a1"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305217","number":305217,"commit":"a51e110b4513825ad9df257c4ab1b2dab2592647"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305158","number":305158,"commit":"6d41b31a555a75523f362208dee136f3febfc550"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304509","number":304509,"commit":"194e4286ce9fc8eb5b82229c6b6e44c6986f8f67"}],"number":306568}
buildkite-pr-comment-->

denar50 committed:
handle the case when toScheduledItem throws an error while updating gaps

denar50 committed:
handle the case when toScheduledItem throws an error while updating gaps

denar50 committed:
add unit tests for clipDateInterval

dplumlee commented on x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/bulk_actions/fetch_rules_by_query_or_ids.ts:
```diff
@@ -7,39 +7,39 @@
         }
-        return rule;
-      },
-      abortSignal,
```
@xcrzx I seem to remember you doing work on bulk action requests timing out, do you think removing the `abortSignal` here is problematic?

dplumlee replied:
Yeah, I remember we had issues with the route timing out in the past but I guess using the rulesClient path directly should subvert those issues 👍

dplumlee reviewed (commented):
Rule management changes look pretty good to me @denar50, looks like the bulk delete tests are failing still and I want to make sure of the timeout signal thing before approving

denar50 committed:
Handle the case when bulkGetRules does not return a summary when an exception is thrown

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307126)
* Commit: b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/043ebe521e43b95d70dba1785096af7962ba63e1) | [after](https://github.com/elastic/kibana/commit/b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76) | diff |
| --- | --- | --- | --- |
| `alerting` | 53 | 54 | +1 |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/043ebe521e43b95d70dba1785096af7962ba63e1) | [after](https://github.com/elastic/kibana/commit/b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76) | diff |
| --- | --- | --- | --- |
| `alerting` | 92 | 93 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/043ebe521e43b95d70dba1785096af7962ba63e1) | [after](https://github.com/elastic/kibana/commit/b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76) | diff |
| --- | --- | --- | --- |
| `alerting` | 99 | 100 | +1 |


</details>


### History
* :broken_heart: [Build #306624](https://buildkite.com/elastic/kibana-pull-request/builds/306624) failed 7bfb45e06e9df6c4b98713566c3c4a43188b3be6
* :broken_heart: [Build #306595](https://buildkite.com/elastic/kibana-pull-request/builds/306595) failed bea03583f0088b8d91478921216a911ff79bd8d2
* :broken_heart: [Build #306568](https://buildkite.com/elastic/kibana-pull-request/builds/306568) failed 1cd081e45c559a46db941163be901a94d93053f4
* :broken_heart: [Build #306527](https://buildkite.com/elastic/kibana-pull-request/builds/306527) failed 37ae9b88b481ac69689b5024ccdeb45e191fb7b7
* :green_heart: [Build #305932](https://buildkite.com/elastic/kibana-pull-request/builds/305932) succeeded 5dd6de38d86fb038a72cab32b62a28f38e7ef4a1

cc @denar50

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307126","number":307126,"commit":"b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306624","number":306624,"commit":"7bfb45e06e9df6c4b98713566c3c4a43188b3be6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306595","number":306595,"commit":"bea03583f0088b8d91478921216a911ff79bd8d2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306568","number":306568,"commit":"1cd081e45c559a46db941163be901a94d93053f4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306527","number":306527,"commit":"37ae9b88b481ac69689b5024ccdeb45e191fb7b7"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305932","number":305932,"commit":"5dd6de38d86fb038a72cab32b62a28f38e7ef4a1"}],"number":307126}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307398)
* Commit: 02ec4e29c152ca8989655e00d92b7caf88b133b4

### Failed CI Steps
* [Defend Workflows Cypress Tests on Serverless #4](https://buildkite.com/elastic/kibana-pull-request/builds/307398#01975ef1-c6c4-42dd-b784-21178860f607)

### History
* :green_heart: [Build #307126](https://buildkite.com/elastic/kibana-pull-request/builds/307126) succeeded b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76
* :broken_heart: [Build #306624](https://buildkite.com/elastic/kibana-pull-request/builds/306624) failed 7bfb45e06e9df6c4b98713566c3c4a43188b3be6
* :broken_heart: [Build #306595](https://buildkite.com/elastic/kibana-pull-request/builds/306595) failed bea03583f0088b8d91478921216a911ff79bd8d2
* :broken_heart: [Build #306568](https://buildkite.com/elastic/kibana-pull-request/builds/306568) failed 1cd081e45c559a46db941163be901a94d93053f4
* :broken_heart: [Build #306527](https://buildkite.com/elastic/kibana-pull-request/builds/306527) failed 37ae9b88b481ac69689b5024ccdeb45e191fb7b7
* :green_heart: [Build #305932](https://buildkite.com/elastic/kibana-pull-request/builds/305932) succeeded 5dd6de38d86fb038a72cab32b62a28f38e7ef4a1

cc @denar50

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307398","number":307398,"commit":"02ec4e29c152ca8989655e00d92b7caf88b133b4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307126","number":307126,"commit":"b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306624","number":306624,"commit":"7bfb45e06e9df6c4b98713566c3c4a43188b3be6"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306595","number":306595,"commit":"bea03583f0088b8d91478921216a911ff79bd8d2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306568","number":306568,"commit":"1cd081e45c559a46db941163be901a94d93053f4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306527","number":306527,"commit":"37ae9b88b481ac69689b5024ccdeb45e191fb7b7"}],"number":307398}
buildkite-pr-comment-->

ymao1 commented on x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/update/calculate_gaps_state.ts:
```diff
@@ -40,9 +41,10 @@ export const calculateGapStateFromAllBackfills = async ({
     if ('error' in backfill) {
       continue;
     }
+    const scheduledItems = backfill?.schedule.map(toScheduledItem) ?? [];
```
do you need to catch any errors from `toScheduledItem` here?

denar50 replied:
Good catch! I have just added exception handling logic for this 🙏

ymao1 reviewed (approved):
LGTM! Left a few small comments.

denar50 committed:
Add BulkGet read operation and audit action

denar50 committed:
handle exceptions thrown by toScheduledItem

denar50 commented on x-pack/platform/plugins/shared/alerting/server/lib/rule_gaps/update/update_gaps.ts:
```diff
@@ -36,40 +39,45 @@ interface UpdateGapsParams {
   }
 
-  if (hasFailedBackfillTask || !backfillSchedule || shouldRefetchAllBackfills) {
+  if (hasFailedBackfillTask || !scheduledItems || shouldRefetchAllBackfills) {
```
Agree. Fixed!

dplumlee reviewed (approved):
LGTM @denar50, did some stress tests for the bulk enable route using the `ids` param with 10k rule ids and wasn't able to see any issues, noticeable update in speed 🚀

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/307673)
* Commit: 46048be02566690be477d52cee9d01972bd43603

### Failed CI Steps
* [Serverless Detection Engine - Exceptions - Security Solution Cypress Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/307673#019762a2-c79d-4202-8d3e-3bf9964ac6bb)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/307673#019762a2-c79d-4202-8d3e-3bf9964ac6bb) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/307673/jobs/019762a2-c79d-4202-8d3e-3bf9964ac6bb/artifacts/019762c3-4338-4060-9318-d4d5b199ae19) Serverless Detection Engine - Exceptions - Security Solution Cypress Tests #<span></span>3 / Add exception using data views from rule details Creates an exception item and close all matching alerts Creates an exception item and close all matching alerts

### History
* :broken_heart: [Build #307620](https://buildkite.com/elastic/kibana-pull-request/builds/307620) failed 129fb1473b0371d179b7bf1ac0a270c7da2be38d
* :broken_heart: [Build #307508](https://buildkite.com/elastic/kibana-pull-request/builds/307508) failed e1ceacdf5e5d7eb46193930ca692efaa37f00a28
* :yellow_heart: [Build #307398](https://buildkite.com/elastic/kibana-pull-request/builds/307398) was flaky 02ec4e29c152ca8989655e00d92b7caf88b133b4
* :green_heart: [Build #307126](https://buildkite.com/elastic/kibana-pull-request/builds/307126) succeeded b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76
* :broken_heart: [Build #306624](https://buildkite.com/elastic/kibana-pull-request/builds/306624) failed 7bfb45e06e9df6c4b98713566c3c4a43188b3be6

cc @denar50

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307673","number":307673,"commit":"46048be02566690be477d52cee9d01972bd43603"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307620","number":307620,"commit":"129fb1473b0371d179b7bf1ac0a270c7da2be38d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307508","number":307508,"commit":"e1ceacdf5e5d7eb46193930ca692efaa37f00a28"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307398","number":307398,"commit":"02ec4e29c152ca8989655e00d92b7caf88b133b4"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307126","number":307126,"commit":"b7f699d14fd333a2a3a50d1a32a7ae9438ac8d76"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306624","number":306624,"commit":"7bfb45e06e9df6c4b98713566c3c4a43188b3be6"}],"number":307673}
buildkite-pr-comment-->

denar50 committed:
add bulkGet to read operations

denar50 committed:
fix api integration tests

denar50 committed:
fix integration tests

azasypkin reviewed (approved):
The change LGTM. It looks like you still need to update a few security-related tests to make CI happy, but approving to unblock you.

denar50 committed:
Merge branch 'main' into security-team-10688-optimize-update-gaps

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/308095)
* Commit: d03deb28eac31f82c4350086e5febad16d95d8cc

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/22e6f78b843d8fc6a4504e3baae065340323f882) | [after](https://github.com/elastic/kibana/commit/d03deb28eac31f82c4350086e5febad16d95d8cc) | diff |
| --- | --- | --- | --- |
| `alerting` | 53 | 54 | +1 |

<details><summary>Unknown metric groups</summary>

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/22e6f78b843d8fc6a4504e3baae065340323f882) | [after](https://github.com/elastic/kibana/commit/d03deb28eac31f82c4350086e5febad16d95d8cc) | diff |
| --- | --- | --- | --- |
| `alerting` | 96 | 97 | +1 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/22e6f78b843d8fc6a4504e3baae065340323f882) | [after](https://github.com/elastic/kibana/commit/d03deb28eac31f82c4350086e5febad16d95d8cc) | diff |
| --- | --- | --- | --- |
| `alerting` | 103 | 104 | +1 |


</details>


### History
* :broken_heart: [Build #308017](https://buildkite.com/elastic/kibana-pull-request/builds/308017) failed 96ca64e81ac115df88bf319c8c427518ae0dcfca
* :broken_heart: [Build #307960](https://buildkite.com/elastic/kibana-pull-request/builds/307960) failed 49c4925435c87f51077516613be8aa0021154765
* :broken_heart: [Build #307856](https://buildkite.com/elastic/kibana-pull-request/builds/307856) failed e180efd1ecafcb9f8ff66bae971a9ed8fb1274b5
* :broken_heart: [Build #307745](https://buildkite.com/elastic/kibana-pull-request/builds/307745) failed 0255e7f4c5286e7fb150c77d3a6b73d8bdaf6bba
* :broken_heart: [Build #307673](https://buildkite.com/elastic/kibana-pull-request/builds/307673) failed 46048be02566690be477d52cee9d01972bd43603

cc @denar50

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308095","number":308095,"commit":"d03deb28eac31f82c4350086e5febad16d95d8cc"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/308017","number":308017,"commit":"96ca64e81ac115df88bf319c8c427518ae0dcfca"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307960","number":307960,"commit":"49c4925435c87f51077516613be8aa0021154765"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307856","number":307856,"commit":"e180efd1ecafcb9f8ff66bae971a9ed8fb1274b5"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307745","number":307745,"commit":"0255e7f4c5286e7fb150c77d3a6b73d8bdaf6bba"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/307673","number":307673,"commit":"46048be02566690be477d52cee9d01972bd43603"}],"number":308095}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 8.19

https://github.com/elastic/kibana/actions/runs/15627112699

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 222158` locally
cc: @denar50

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 222158` locally
cc: @denar50

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 222158` locally
cc: @denar50

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 222158` locally
cc: @denar50

nkhristinin commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/224660">](https://github.com/elastic/kibana/pull/224660)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.1 -->
